src/base/static/js/views/place-detail-view.js

```diff
@@ -62,42 +143,315 @@
 
+      // Augment the data with any draft changes saved to localstorage
+      if (this.isEditingToggled &&
+          Util.localstorage.get(this.LOCALSTORAGE_KEY)) {
+        
+        this.isModified = true;
+        data.isModified = true;
+        _.extend(data, Util.localstorage.get(this.LOCALSTORAGE_KEY));
+      }  
+
       this.$el.html(Handlebars.templates['place-detail'](data));
 
       // Render the view as-is (collection may have content already)
       this.$('.survey').html(this.surveyView.render().$el);
-      // Fetch for submissions and automatically update the element
-      this.model.submissionSets[this.surveyType].fetchAllPages();
 
       this.$('.support').html(this.supportView.render().$el);
       // Fetch for submissions and automatically update the element
       this.model.submissionSets[this.supportType].fetchAllPages();
 
       this.delegateEvents();
+      this.surveyView.delegateEvents();
 
       $("#content article").animate({ scrollTop: 0 }, "fast");
+      
+      // initialize datetime picker, if relevant
+      $('#datetimepicker').datetimepicker({ formatTime: 'g:i a' });
+
+      if (this.isEditingToggled) {
+        $("#toggle-editor-btn").addClass("btn-depressed");
+        $(".promotion, .place-submission-details, .survey-header, .reply-link, .response-header")
+          .addClass("faded");
+
+        // detect changes made to non-Quill form elements
+        $(this.watchFields).on("keyup change", function(e) {
+          if (e.type === "change") {
+            self.onModified();
+          } else if ((e.keyCode >= 48 && e.keyCode <= 57) || // 0-9 (also shift symbols)
+              (e.keyCode >= 65 && e.keyCode <= 90) || // a-z (also capital letters)
+              (e.keyCode === 8) || // backspace key
+              (e.keyCode === 46) || // delete key
+              (e.keyCode === 32) || // spacebar
+              (e.keyCode >= 186 && e.keyCode <= 222)) { // punctuation
+            
+            self.onModified();
+          }
+        });
+
+        $(".rich-text-field").each(function() {
+          new RichTextEditorView({
+            el: $(this).get(0),
+            model: self.model,
+            placeDetailView: self,
+            fieldName: $(this).attr("name"),
+            fieldId: $(this).attr("id")
+          });
+        });
+      }
 
       return this;
     },
+
+    tearDown: function() {
+      this.options.router.off("route", this.tearDown, this);
+      this.isEditingToggled = false;
+      this.isModified = false;
+    },
+
+    onModified: function() {
+      this.isModified = true;
+      $("#update-place-model-btn").addClass("isModified");
+      $(this.watchFields).off("keyup change");
+    },
+
     remove: function() {
       // Nothing yet
     },
 
     onChange: function() {
       this.render();
+    },
+
+    // called by the router
+    onCloseWithUnsavedChanges: function() {
+      // if (confirm("You have unsaved changes. Proceed?")) {
```

Comment:
So if someone made some changes, then left the page, what would happen? Would the locally-stored changes get uploaded in the background? If not, that doesn't seem like a great experience (without warning, the changes that they made aren't actually saved on the server). This probably merits more design discussion.